mantis-common/src/main/java/io/mantisrx/common/metrics/LoggingMetricsPublisher.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.common.metrics;
+
+import io.mantisrx.shaded.com.google.common.base.Joiner;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This implementation will print the selected metric groups in log.
+ * To use this publisher override config "mantis.metricsPublisher.class" in the master-*.properties file.
+ */
+@Slf4j
+public class LoggingMetricsPublisher extends MetricsPublisher {
```

Comment:
just curious - why is this required? we are anyways publishing these metrics today via spectator - what's the need to log it.

Reply:
If this is only expected to be used for testing purposes, I recommend that you move this class to the test package and make it package private.

Reply:
It is a debug mode but to use this it needs to be configured via the core config (this interface is one of these created in config instead of injection).
Also in OSS domain, I think it's okay to have a logging-based mode option.